akka-remote/src/main/java/akka/remote/artery/compress/TopHeavyHittersScanning.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery.compress;
+
+public class TopHeavyHittersScanning {
+
+  private int _lowestHitterIdx = 0;
+  
+  protected final int[] hashes;
+  protected final Object[] items;
```

Comment:
Moved most things into Scala, the array scanning remains in Java (see benchmark)